src/domHelpers/index.ts

```diff
@@ -1,6 +1,19 @@
 export const isHTML = (str) => {
   const doc = new DOMParser().parseFromString(str, "text/html");
-  return Array.from(doc.body.childNodes).some((node) => node.nodeType === 1);
+  const childNodes = doc.body.childNodes;
+
+  return Array.from(childNodes).some(( node, index ) => {
+    if ( node.nodeType === 1 ) {
+      const nextElemText = childNodes[ index + 1 ]?.textContent;
+
+      // this checks to see if the html is residing within a code block
+      // the next block is a new line, that means the html resides within a code block
+      return !nextElemText ? true :
+        !nextElemText.startsWith( '\n' );
+    }
```

Comment:
can you give more context about the bug and how this change is solving it?

Reply:
so basically this solves the bug where when sometimes the note messes up after refresh:
https://streamable.com/mbxydm

 I found out that it's because the HTML check parses the doc for anything that looks like or is HTML code and treats it a HTML doc, so for the one pasted code blog post, it had HTML within code blocks messing up the format.

The desired behavior is to have anything with HTML in the document to be parsed as HTML EXCEPT for HTML within code blocks. The above check is kind of a hack I found out after a lot of console.logging -> anything in a code block that is detected as HTML will be followed with a new line (\n). So if that happens, the above code ignores that HTML block within the check.

I think it should be fine because new lines for the editor are backslashes (\), so I THINK it shouldn't mess with any notes saved. For old HTML notes -> i did some testing and i think it should be fine too.